### PR TITLE
Adding polling time as Oxidized configuration parameter.

### PR DESCRIPTION
**Pull request by elpatopedraza**
## Pre-Request Checklist

- [x] Passes rubocop code analysis (try `rubocop --auto-correct`)
- [x] Tests added or adapted (try `rake test`)
- [x] Changes are reflected in the documentation
- [x] User-visible changes appended to [CHANGELOG.md](/CHANGELOG.md)

### CODE REVIEW

**Comment by ytti** [comment]
Can you explain the utility of this? When would you like to change this value?

**Comment by elpatopedraza** [comment]
> Can you explain the utility of this? When would you like to change this value?

Hi @ytti, sure I'll answer your questions explaining our current situation:

Seems like our organization has some rate limiter over the network devices my team has to monitor. This means that we can't create more that certain amount of SSH sessions per minute from the same IP address. We are using Oxidized to monitor a couple of network devices, but seems like the backup process is so fast that Oxidized tries to run multiple times the backup process in a single minute, so the rate limiter is blocking some SSH sessions after certain limit and Oxidized is marking those attempts as failed (besides that the rate limiter is triggering some alarms due to exceeded SSH sessions).

Basically this new feature allows to any user define the Oxidized polling time, so the platform starts the backup process and waits for x seconds until start the next iteration.

This fix our issue and would be nice to have it in the next Oxidized version.

I hope this little story is clear for you :)

**Comment by ytti** [comment]
We should only try to fetch every device once per `interval`. But if we don't succeed, we try same box `retries` times back-to-back, before giving up.

This parameter you are changing is about how often we check if we have any work to do. So it shouldn't cause any network connections, if everything has been done.

If we try to hit boxes more often than `interval` * `retries`, then something is off.

**Comment by elpatopedraza** [comment]
> We should only try to fetch every device once per `interval`. But if we don't succeed, we try same box `retries` times back-to-back, before giving up.
> 
> This parameter you are changing is about how often we check if we have any work to do. So it shouldn't cause any network connections, if everything has been done.
> 
> If we try to hit boxes more often than `interval` * `retries`, then something is off.

@ytti I understand your point regarding retries and interval parameters.

Let's suppose we have only 2 network devices registered, the rate limiter only allows 2 SSH sessions per minute and the interval parameter is the default one (3600 seconds). Oxidized runs the first backup iteration successfully (less than a minute)...good so far... Oxidized waits 1 second and starts the second iteration... good so far... waits 1 second and starts the third iteration... at this point our rate limiter starts to trigger issues regarding Oxidized IP and unfortunately this policy cannot be changed by my team because it comes from security policies. So we thought the best approach was to configure this parameter to avoid this.

Please let me know your comments.

**Comment by ytti** [comment]
It shouldn't start the 3rd iteration until 3600s have passed. It should fetch both of the devices back-to-back, then do nothing for 3600s. 
If it does immediately fetch the 1st node again, then something is broken in the code :(

**Comment by elpatopedraza** [comment]
> It shouldn't start the 3rd iteration until 3600s have passed. It should fetch both of the devices back-to-back, then do nothing for 3600s.
> If it does immediately fetch the 1st node again, then something is broken in the code :(

Ok, I'll check this more carefully with the team and I'll back to you. Thanks.

**Comment by ytti** [comment]
Generally what the algorithm tries to do, it tries to ensure that we spread the work over as long period of time as possible. This is on premise the we want predictable CPU use, so we don't have to provision for high peak use, when box is mostly silent. However the 'slowest' we go, is as fast as single thread can go. So if the algorithm determines we can fetch all your devices with single thread, when that single thread still fetches back-to-back all the devices as fast as it can, and if there is time remaining it does nothing.

So if you have very strict SSH limits and the single thread performs too fast, you could exceed those limits. And if you need this sleep time configurable (awkward fix to your problem) I can accept it on some other name like 'loop_sleep' or something. 

The whole reason there is _any sleep_ at all, is to avoid busy loop, when we don't have anything to do, we'd potentially use very much CPU detecting we don't have anything to do rapidly over and over again.

**Comment by mortzu** [comment]
Any news? Can this be closed?
